OpenSkyPlusApi/OpenSkyPlusApi.cs

```diff
@@ -1,722 +1,754 @@
+
+        return _config.AppSettings.ShotConfidence switch
+        {
+            "Forgiving" => averageConfidence >= 0.75f,
+            "Strict" => averageConfidence > 0,
+            _ => averageConfidence >= 0.25f
+        };
+    }
+
+    private void LogShot(ShotData shot)
+    {
+        var sb = new StringBuilder($"{GetShotMode()} Shot Received:");
+        sb.AppendLine("ClubData");
+        sb.AppendLine("----");
+        sb.AppendLine($"\tHead Speed: {shot.Club.HeadSpeed}");
+        sb.AppendLine($"\tHead Speed Confidence: {shot.Club.HeadSpeedConfidence}");
+        sb.AppendLine("LaunchData");
+        sb.AppendLine("------");
+        sb.AppendLine($"\tHorizontal Angle: {shot.Launch.HorizontalAngle}");
+        sb.AppendLine($"\tHorizontal Angle Confidence: {shot.Launch.HorizontalAngleConfidence}");
+        sb.AppendLine($"\tVertical Angle: {shot.Launch.LaunchAngle}");
+        sb.AppendLine($"\tVertical Angle Confidence: {shot.Launch.LaunchAngleConfidence}");
+        sb.AppendLine($"\tTotal Speed: {shot.Launch.TotalSpeed}");
+        sb.AppendLine($"\tTotal Speed Confidence: {shot.Launch.TotalSpeedConfidence}\n");
+        sb.AppendLine("SpinData");
+        sb.AppendLine("----");
+        sb.AppendLine($"\tSpin Axis: {shot.Spin.SpinAxis}");
+        sb.AppendLine($"\tBackspin: {shot.Spin.Backspin}");
+        sb.AppendLine($"\tSide Spin: {shot.Spin.SideSpin}");
+        sb.AppendLine($"\tTotal Spin spin: {shot.Spin.TotalSpin}");
+        sb.AppendLine($"\tMeasurement Confidence: {shot.Spin.MeasurementConfidence}");
+        _logger.LogInfo(sb.ToString());
+    }
+
+    public void LogToOpenSkyPlus(string message, object level)
+    {
+        try
+        {
+            switch ((LogLevels)level)
+            {
+                case LogLevels.Info:
+                    _logger.LogInfo(message);
+                    break;
+                case LogLevels.Warning:
+                    _logger.LogWarning(message);
+                    break;
+                case LogLevels.Debug:
+                    _logger.LogDebug(message);
+                    break;
+                case LogLevels.Error:
+                default:
+                    _logger.LogError(message);
+                    break;
+            }
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"Failed to translate logging request to {_logger.GetType().Name}: {ex}");
+        }
+    }
+
+    /// <summary>
+    ///     Logs all the raw data from a shot.
+    ///     Useful for understanding what we are looking at. We probably don't need it anymore,
+    ///     but it was annoying to write, so I'm leaving it here in case its needed later.
+    /// </summary>
+    private void ShotDump(object shot)
+    {
+        var Shot = shot.GetType();
+
+        var FDDDFMHPPEP = Shot.GetField("FDDDFMHPPEP",
```

Comment:
object is technically more correct here as we know the exact type we receive from reflection (which is just an object)